var/spack/repos/builtin/packages/pumi/package.py

```diff
@@ -109,15 +109,15 @@ def test(self):
         if self.spec.version <= Version("2.2.6"):
             return
         exe = "uniform"
-        options = ["../testdata/pipe.dmg", "../testdata/pipe.smb", "pipe_unif.smb"]
+        options = ["../share/testdata/pipe.dmg", "../share/testdata/pipe.smb", "pipe_unif.smb"]
         expected = "mesh pipe_unif.smb written"
         description = "testing pumi uniform mesh refinement"
         self.run_test(exe, options, expected, purpose=description, work_dir=self.prefix.bin)
```

Comment:
Stand-alone tests should not be run from the prefix due to issues with permissions in read-only installations. This can be avoided by either providing the full path to the files in the prefix or by copying the test data to the test stage and using the full path to executable(s).  For example,

```suggestion
        data_dir = self.prefix.share.testdata
        exe = self.prefix.bin.uniform
        options = [
            join_path(data_dir, "pipe.dmg"),
            join_path(data_dir, "pipe.smb"),
            join_path(self.prefix.bin, "pipe_unif.smb"),
        ]
        expected = "pipe_unif.smb written"
        description = "testing pumi uniform mesh refinement"
        self.run_test(exe, options, expected, purpose=description)
```